### PR TITLE
feat (worker): worker artifacts cmd

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

File: engine/worker/cmd_artifacts.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		resp, errDo := client.Do(req)
+		if errDo != nil {
+			sdk.Exit("cannot post worker artifacts (Do): %s\n", errDo)
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode >= 300 {
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				sdk.Exit("cannot artefact artifacts HTTP %v\n", err)
```
Comment (sguiheux) [inline engine/worker/cmd_artifacts.go]: rework message

File: sdk/workflow_run.go [hunk 2]
```diff
@@ -144,12 +144,12 @@ type WorkflowNodeRunArtifact struct {
...
 	ID                int64     `json:"id" db:"id"`
-	Name              string    `json:"name" db:"name"`
-	Tag               string    `json:"tag" db:"tag"`
+	Name              string    `json:"name" db:"name" cli:"name"`
+	Tag               string    `json:"tag" db:"tag" cli:"tag"`
 	DownloadHash      string    `json:"download_hash" db:"download_hash"`
 	Size              int64     `json:"size,omitempty" db:"size"`
 	Perm              uint32    `json:"perm,omitempty" db:"perm"`
-	MD5sum            string    `json:"md5sum,omitempty" db:"md5sum"`
+	MD5sum            string    `json:"md5sum,omitempty" db:"md5sum" cli:"md4sum"`
```
Comment (sguiheux) [inline sdk/workflow_run.go]: md4?

File: engine/worker/cmd_artifacts.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		Use:   "artifacts",
+		Short: "worker artifacts [--workflow=<workflow-name>] [--number=<run-number>] [--tag=<tag>] [--pattern=<pattern>]",
+		Long: `
+Inside a job, you can list artifacts of a workflow:
+
+	worker artifacts --pattern="files.*.yml"
+
+	#theses two commands have the same result:
+	worker artifacts
+	worker artifacts --workflow={{.cds.workflow}} --number={{.cds.run.number}}
```
Comment (sguiheux) [inline engine/worker/cmd_artifacts.go]: don't understand why we need the second line

Comment (yesnault) [reply engine/worker/cmd_artifacts.go]: coherence with worker download

**Pull request merged**